src/AppInstallerCommonCore/Deployment.cpp

```diff
@@ -66,49 +66,108 @@ namespace AppInstaller::Deployment
 
             return S_OK;
         }
+
+        bool ShouldUseReputationCheck(const Options& options)
+        {
+            return options.ExpectedDigests.empty() && !options.SkipReputationCheck;
+        }
+
+        IAsyncOperationWithProgress<DeploymentResult, DeploymentProgress> StartAddPackage(PackageManager& packageManager, const winrt::Windows::Foundation::Uri& uri, const Options& options)
+        {
+            if (!options.ExpectedDigests.empty())
+            {
+                // Must use API that supports digests
+                THROW_WIN32_IF(ERROR_NOT_SUPPORTED, !IsExpectedDigestsSupported());
+
+                AddPackageOptions addPackageOptions;
+
+                for (const auto& digest : options.ExpectedDigests)
+                {
+                    addPackageOptions.ExpectedDigests().Insert(Uri{ Utility::ConvertToUTF16(digest.first) }, digest.second);
+                }
+
+                return packageManager.AddPackageByUriAsync(uri, addPackageOptions);
+            }
+            else if (options.SkipReputationCheck)
+            {
+                return packageManager.AddPackageAsync(
+                    uri,
+                    nullptr, /*dependencyPackageUris*/
+                    DeploymentOptions::None,
+                    nullptr, /*targetVolume*/
+                    nullptr, /*optionalAndRelatedPackageFamilyNames*/
+                    nullptr, /*optionalPackageUris*/
+                    nullptr /*relatedPackageUris*/);
+            }

```

Comment:
Is there a reason to have these two branches separate? I would find it easier to understand if we had as few different calls to `*AddPackage*Async()`
Only reason I can think of is if `AddPackageByUriAsync()` wasn't available downlevel, but it goes back to 19041 which I think is the oldest supported version

Reply:
I don't want to remove it until we change the minimum version for the shipping package:
```
<TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.17763.0" MaxVersionTested="10.0.22621.0"/>
```